### PR TITLE
Fix cache missing metrics

### PR DESCRIPTION
**Pull request by zhangchiqing**
see comments inline

### CODE REVIEW

File: storage/badger/cache.go [hunk 1]
```diff
@@ -94,12 +94,14 @@ func (c *Cache) Get(key interface{}) func(*badger.Txn) (interface{}, error) {
 		}
 
 		// get it from the database
-		c.metrics.CacheMiss(c.resource)
```
Comment (zhangchiqing) [inline storage/badger/cache.go:97]: There are 2 cases when cache was missed:
1. the queried item was not found in database
2. the queried item exists in database.

We now use different metrics to distinguish them

File: module/metrics/cache.go [hunk 2]
```diff
@@ -33,11 +34,19 @@ func NewCacheCollector(chain flow.ChainID) *CacheCollector {
 			ConstLabels: prometheus.Labels{LabelChain: chain.String()},
 		}, []string{LabelResource}),
 
+		notfounds: promauto.NewCounterVec(prometheus.CounterOpts{
+			Name:        "notfounds_total",
+			Namespace:   namespaceStorage,
+			Subsystem:   subsystemCache,
+			Help:        "the number of times the queried item was not found in either cache or database",
+			ConstLabels: prometheus.Labels{LabelChain: chain.String()},
```
Comment (jordanschalm) [inline module/metrics/cache.go:42]: Do you know why we use the `chain` label here? Seems unrelated to the storage caches. 🤔

File: storage/badger/cache.go [hunk 3]
```diff
@@ -94,12 +94,14 @@ func (c *Cache) Get(key interface{}) func(*badger.Txn) (interface{}, error) {
...
 		resource, err := c.retrieve(key)(tx)
 		if err != nil {
+			c.metrics.CacheNotFound(c.resource)
```
Comment (jordanschalm) [inline storage/badger/cache.go:99]:
```suggestion
		if errors.Is(err, storage.ErrNotFound) {
			c.metrics.CacheNotFound(c.resource)
		}
		if err != nil {
```

**Review by jordanschalm — APPROVED** [review]
👍  Good idea. Just one suggestion

File: storage/badger/cache.go [hunk 4]
```diff
@@ -94,12 +94,16 @@ func (c *Cache) Get(key interface{}) func(*badger.Txn) (interface{}, error) {
 		}
 
 		// get it from the database
-		c.metrics.CacheMiss(c.resource)
 		resource, err := c.retrieve(key)(tx)
+		if errors.Is(err, storage.ErrNotFound) {
```
Comment (m4ksio) [inline storage/badger/cache.go:98]: Its my being picky - we can do this check only after `err` is not `nil`, therefore saving one or two CPU cycles - I think this happen often enough to be worth it

Comment (zhangchiqing) [reply storage/badger/cache.go]: makes sense. updated

**Review by m4ksio — APPROVED** [review]
